File: leetcode/lc-us/cpp/2395.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution {
public:
    bool findSubarrays(vector<int> &nums) {
        int n = (int) nums.size();
        unordered_set<int> S;
        for (int i = 0; i < n - 1; i++) {
            int s = nums[i] + nums[i + 1];
            if (S.count(s)) return true;
            S.insert(s);
        }
        return false;
    }
};

File: leetcode/lc-us/rust/2395.rs
use std::collections::HashSet;

impl Solution {
    pub fn find_subarrays(nums: Vec<i32>) -> bool {
        let mut hash: HashSet<i32> = HashSet::new();
        for i in 1..nums.len() {
            let diff = nums[i] + nums[i - 1];
            if hash.contains(&diff) {
                return true;
            }
            hash.insert(diff);
        }
        false
    }
}